content/documents/implementing-single-sign-on-with-google-idp.mdx

```diff
@@ -0,0 +1,413 @@
+  AdditionalIdpScopes: []
+})
+.catch((err:OrderCloudError) => console.log(err));
+console.log(openIdConnect);`,
+csharp: `using OrderCloud.SDK;\n
+var client = new OrderCloudClient(...);\n
+await client.AuthenticateAsync();\n
+OpenIdConnect response = await client.OpenIdConnects.CreateAsync(new OpenIdConnect {
+  ID: "google",
+  OrderCloudApiClientID: "CLIENT_ID_FROM_STEP_1.4",
+  ConnectClientID: "GOOGLE_CLIENT_ID_HERE",
+  ConnectClientSecret: "GOOGLE_CLIENT_SECRET_HERE",
+  AppStartUrl: "http://localhost:4451?token={0}&idpToken={1}",
+  AuthorizationEndpoint: "https://accounts.google.com/o/oauth2/v2/auth",
+  TokenEndpoint: "https://oauth2.googleapis.com/token",
+  UrlEncoded: false,
+  CallSyncUserIntegrationEvent: true,
+  IntegrationEventID: "openidconnect",
+  AdditionalIdpScopes: []
+});`}}
+/>
+
+| OrderCloud Property     | Description                                                                                                                                                                                                                |
+    |-------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+    | `ID`                    | Unique identifier for the connect client config                                                                                                                                                                            |
+    | `OrderCloudApiClientID` | This is the clientID (on OrderCloud) that wants to authenticate via OpenID Connect                                                                                                                                         |
+    | `ConnectClientID`       | This is the clientID of the identify provider (in this case Google)                                                                                                                                                        |
+    | `ConnectClientSecret`   | This is the clientSecret of the identity provider (in this case Google)                                                                                                                                                    |
+    | `AppStartUrl`           | This is where the user will be redirected to after successful authentication via OpenID Connect. The parameter `{0}` represents the token. The parameter `{1} represents the IDP token, and the parameter {2} which is not used here represents the app start path used for deep linking`                                                                                  |
+    | `AuthorizationEndpoint` | Defined by the OpenID provider (in this case Google). It is the endpoint OrderCloud will redirect user to in order to validate credentials.                                                                                |
```

Comment:
It is the endpoint OrderCloud will redirect the* user to in order to validate credentials.